File: study/lua/src/app/ui/PopupChangeJade.lua
--弹出界面
--物品信息弹框
--点击物品时，弹出
local PopupBase = require("app.ui.PopupBase")
local PopupChangeJade = class("PopupChangeJade", PopupBase)
local Path = require("app.utils.Path")

local TypeConvertHelper = require("app.utils.TypeConvertHelper")
local UserDataHelper = require("app.utils.UserDataHelper")

function PopupChangeJade:ctor(title, slot, jadeUnitData, equipUnitData, isRed, callback, type)
	--
	self._title = title or Lang.get("common_title_item_info")
	self._callback = callback
	self._slot = slot
	self._jadeUnitData = jadeUnitData
	self._equipUnitData = equipUnitData
	self._isRed = isRed
	self._type = type and type or FunctionConst.FUNC_EQUIP_TRAIN_TYPE3
	--
	local resource = {
		file = Path.getCSB("PopupChangeJade", "common"),
		binding = {
			_btnChange = {
				events = {{event = "touch", method = "onBtnChange"}}
			},
			_btnUnload = {
				events = {{event = "touch", method = "onBtnUnload"}}
			}
		}
	}
	PopupChangeJade.super.ctor(self, resource, true)
end

--
function PopupChangeJade:onCreate()
	-- button
	self._btnChange:setString(Lang.get("equipment_choose_jade_cell_btn4"))
	self._btnChange:switchToHightLight()
	self._btnChange:showRedPoint(self._isRed)
	self._btnUnload:setString(Lang.get("equipment_choose_jade_cell_btn3"))
	self._btnUnload:switchToNormal()
	self._commonNodeBk:addCloseEventListener(handler(self, self.onBtnCancel))
	self._commonNodeBk:setTitle(self._title)
	self._commonNodeBk:hideCloseBtn()
end

function PopupChangeJade:_onInit()
end

function PopupChangeJade:onEnter()
	local config = self._jadeUnitData:getConfig()
	self._itemIcon:updateUI(config.id)
	self._itemIcon:setTouchEnabled(false)
	self._itemName:setColor(Colors.getColor(config.color))
	self._itemName:setString(config.name)

	self._itemDesc = cc.Label:createWithTTF("", Path.getCommonFont(), 20)
	self._itemDesc:setColor(Colors.BRIGHT_BG_TWO)
	self._itemDesc:setWidth(260)
	self._itemDesc:setAnchorPoint(cc.p(0, 1))
	self._scrollView:addChild(self._itemDesc)
	self._itemDesc:setString(config.description)
	local height = self._itemDesc:getContentSize().height
	local scrollHeight = self._scrollView:getContentSize().height
	if height > scrollHeight then		
		self._itemDesc:setPosition(cc.p(2, height))
		self._scrollView:setInnerContainerSize(cc.size(320, height + 5))
	else
		self._itemDesc:setPosition(cc.p(2, scrollHeight))
		self._scrollView:setInnerContainerSize(cc.size(320, scrollHeight))
	end

end

function PopupChangeJade:onExit()
end

--
function PopupChangeJade:onBtnChange()
	local EquipJadeHelper = require("app.scene.view.equipmentJade.EquipJadeHelper")
	local list = EquipJadeHelper.getEquipJadeListByWear(self._slot, self._jadeUnitData, self._equipUnitData, false, self._type)
	if #list > 0 then
		EquipJadeHelper.popupChooseJadeStone(self._slot, self._jadeUnitData, self._equipUnitData, self._callback, true, self._type)
	else
		G_Prompt:showTip(Lang.get("equipment_choose_jade_tips"))
	end
	self:close()
end

function PopupChangeJade:onBtnUnload()
	if self._callback then
		self._callback(self._slot, 0)
	end
	self:close()
end

function PopupChangeJade:onBtnCancel()
	self:close()
end

return PopupChangeJade

File: study/ts/script/ui/PopupChangeJade.ts
const { ccclass, property } = cc._decorator;

import CommonNormalMiniPop from './component/CommonNormalMiniPop'
import CommonButtonSwitchLevel0 from './component/CommonButtonSwitchLevel0';
import CommonJadeIcon from './component/CommonJadeIcon';
import { Lang } from '../lang/Lang';
import { handler } from '../utils/handler';
import { Colors, G_Prompt } from '../init';
import { EquipJadeHelper } from '../scene/view/equipmentJade/EquipJadeHelper';
import PopupBase from './PopupBase';

@ccclass
export default class PopupChangeJade extends PopupBase {

    public static path = 'common/PopupChangeJade';

    @property({
        type: CommonNormalMiniPop,
        visible: true
    })
    _commonNodeBk: CommonNormalMiniPop = null;
    @property({
        type: CommonButtonSwitchLevel0,
        visible: true
    })
    _btnChange: CommonButtonSwitchLevel0 = null;
    @property({
        type: CommonButtonSwitchLevel0,
        visible: true
    })
    _btnUnload: CommonButtonSwitchLevel0 = null;

    @property({
        type: CommonJadeIcon,
        visible: true
    })
    _itemIcon: CommonJadeIcon = null;
    @property({
        type: cc.Label,
        visible: true
    })
    _itemDesc: cc.Label = null;
    @property({
        type: cc.Label,
        visible: true
    })
    _itemName: cc.Label = null;
    _title: any;
    _callback: any;
    _slot: any;
    _jadeUnitData: any;
    _equipUnitData: any;
    _isRed: any;

    ctor(title, slot, jadeUnitData, equipUnitData, isRed, callback,type) {
        this._title = title || Lang.get('common_title_item_info');
        this._callback = callback;
        this._slot = slot;
        this._jadeUnitData = jadeUnitData;
        this._equipUnitData = equipUnitData;
        this._isRed = isRed;
        this._isClickOtherClose = true;
    }
    onCreate() {
        this._btnChange.setString(Lang.get('equipment_choose_jade_cell_btn4'));
        this._btnChange.switchToHightLight();
        this._btnChange.showRedPoint(this._isRed);
        this._btnUnload.setString(Lang.get('equipment_choose_jade_cell_btn3'));
        this._btnUnload.switchToNormal();
        this._commonNodeBk.addCloseEventListener(handler(this, this.onBtnCancel));
        this._commonNodeBk.setTitle(this._title);
        this._commonNodeBk.hideCloseBtn();
        this._btnChange.addClickEventListenerEx(handler(this, this.onBtnChange));
        this._btnUnload.addClickEventListenerEx(handler(this, this.onBtnUnload));
    }
    _onInit() {
    }
    onEnter() {
        var config = this._jadeUnitData.getConfig();
        this._itemIcon.updateUI(config.id);
        this._itemIcon.setTouchEnabled(false);
        this._itemName.string = (config.name);
        this._itemName.node.color = (Colors.getColor(config.color));
        this._itemDesc.string = (config.description);
    }
    onExit() {
    }
    onBtnChange() {
        var list = EquipJadeHelper.getEquipJadeListByWear(this._slot, this._jadeUnitData, this._equipUnitData, false);
        if (list.length > 0) {
            EquipJadeHelper.popupChooseJadeStone(this._slot, this._jadeUnitData, this._equipUnitData, this._callback, true);
        } else {
            G_Prompt.showTip(Lang.get('equipment_choose_jade_tips'));
        }
        this.close();
    }
    onBtnUnload() {
        if (this._callback) {
            this._callback(this._slot, 0);
        }
        this.close();
    }
    onBtnCancel() {
        this.close();
    }

}